wagtail/admin/tests/test_edit_handlers.py

```diff
@@ -983,22 +940,21 @@ def test_autodetect_page_type(self):
         self.assertIn(expected_js, result)
 
     def test_target_models(self):
```

Comment:
Before this PR, 

if you have a FK to a specific model, `target_models` includes it.

e.g. for

```python
    page = models.ForeignKey(
        "tests.EventPage", help_text="more help text", on_delete=models.CASCADE
    )
```

the following test is successful

```python
    def test_target_models_on_fk_with_specific_model(self):
        result = PageChooserPanel(
            'page',
        ).bind_to(model=EventPageChooserModel).target_models()
        self.assertEqual(result, [EventPage])
```


after this PR:

```python
    def test_target_models_on_fk_with_specific_model(self):
        panel = PageChooserPanel("page").bind_to(
            model=EventPageChooserModel
        )
        widget = panel.widget_overrides()["page"]
        self.assertEqual(widget.target_models, [EventPage])
```

fails with:

```
ERROR: test_target_models_on_fk_with_specific_model (wagtail.admin.tests.test_edit_handlers.TestPageChooserPanel)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "wagtail/admin/tests/test_edit_handlers.py", line 953, in test_target_models_on_fk_with_specific_model
    widget = panel.widget_overrides()["page"]
KeyError: 'page'
```

Reply:
This is expected. WagtailAdminModelForm now [has its own rule](https://github.com/wagtail/wagtail/blob/ef2fa2e3944a7844b08aa070552f5046d5433a7f/wagtail/admin/forms/models.py#L66-L68) to select an AdminPageChooser with an appropriate `target_models` argument when it encounters a foreign key to a Page model - as such, it's no longer PageChooserPanel's responsibility to specify an override for that widget (unless PageChooserPanel is configured with an explicit `page_type` or `can_choose_root` argument, which WagtailAdminModelForm won't know about from looking at the ForeignKey alone).

To preserve the intent of that test, I would build the final form and confirm that the expected widget appears there - something like:

    from wagtail.admin.forms.models import WagtailAdminModelForm

    class EventPageChooserForm(WagtailAdminModelForm):
        class Meta:
            model = EventPageChooserModel
            fields = ['page']

    ...

        def test_target_models_on_fk_with_specific_model(self):
            form = EventPageChooserForm()
            widget = form.fields['page'].widget
            self.assertEqual(widget.target_models, [EventPage])

Reply:
Argh, I missed that. Thank you Matthew.

Looks like we'll want to be smarter about in on https://github.com/wagtail/wagtail-localize/blob/main/wagtail_localize/views/edit_translation.py#L266-L280